src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,495 @@
+
+    String EXPECTED_JSON_PATH_VIDEO =
+            "src/test/java/org/prebid/server/analytics/reporter/greenbids/greenbids_analytics_payload_video.json";
+
+    String EXPECTED_JSON_PATH_BANNER_FORMAT_NULL =
+            "src/test/java/org/prebid/server/analytics/reporter/greenbids/greenbids_analytics_payload_no_format.json";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private HttpClient httpClient;
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private PrebidVersionProvider prebidVersionProvider;
+
+    @Captor
+    private ArgumentCaptor<String> jsonCaptor;
+
+    private GreenbidsAnalyticsReporter target;
+
+    private AuctionEvent event;
+
+    private GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+
+    @Before
+    public void setUp() {
+        greenbidsAnalyticsProperties = GreenbidsAnalyticsProperties.builder()
+                .exploratorySamplingSplit(0.9)
+                .analyticsServerVersion("2.2.0")
+                .analyticsServer("http://localhost:8080")
+                .configurationRefreshDelayMs(10000L)
+                .timeoutMs(100000L)
+                .build();
+
+        target = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient,
+                clock,
+                prebidVersionProvider);
+
+        jsonCaptor = ArgumentCaptor.forClass(String.class);
+    }
+
+    @Test
+    public void shouldReceiveValidResponseOnAuctionContextForBanner() throws IOException {
+        // given
+        final Banner banner = givenBanner();
+
+        final ObjectNode prebidJsonNodes = mapper.valueToTree(
+                singletonMap("gpid", TextNode.valueOf("gpidvalue")));
+
+        final Imp imp = Imp.builder()
+                .id("adunitcodevalue")
+                .ext(prebidJsonNodes)
+                .banner(banner)
+                .build();
+
+        final AuctionContext auctionContext = givenAuctionContext(imp);
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final HttpClientResponse mockResponse = mock(HttpClientResponse.class);
+        when(mockResponse.getStatusCode()).thenReturn(202);
+        when(httpClient.post(anyString(), any(MultiMap.class), jsonCaptor.capture(), anyLong()))
+                .thenReturn(Future.succeededFuture(mockResponse));
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        final String capturedJson = jsonCaptor.getValue();
+        final ObjectNode capturedJsonNode = (ObjectNode) mapper.readTree(capturedJson);
+        capturedJsonNode.put("greenbidsId", "testGreenbidsId");
+        capturedJsonNode.put("billingId", "testBillingId");
```

Comment:
I've added modifs in `capturedJsonNode` because in `processEvent` these Ids are generated as random UUIDs (ex: `"5a0e8768-4d98-4101-a3dc-d10fd9b12b41"`) and I force them in test to be constant to be able to compare expected and actual jsons

```java
final String greenbidsId = UUID.randomUUID().toString();
final String billingId = UUID.randomUUID().toString();
```